### PR TITLE
Visit mapping and list constructors

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
With this change, the following case will be modeled as a new data model.

```ballerina
function testJson(int amount) returns error? {
    json newData = {value: check calculate(amount)};
}
```

### CODE REVIEW

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java [hunk 1]
```diff
@@ -487,6 +489,21 @@ public void visit(CheckExpressionNode checkExpressionNode) {
         nodeBuilder.codedata().lineRange(checkExpressionNode);
     }
 
+    @Override
+    public void visit(MappingConstructorExpressionNode mappingCtrExprNode) {
```
Comment (hasithaa) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java:493]: Shall we do this for ListConstructor too?

Comment (KavinduZoysa) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java:494]: Added

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java [hunk 2]
```diff
@@ -487,6 +490,36 @@ public void visit(CheckExpressionNode checkExpressionNode) {
...
+        if (kind == SyntaxKind.LOCAL_VAR_DECL || kind == SyntaxKind.MODULE_VAR_DECL) {
+            startNode(FlowNode.Kind.NEW_DATA)
+                    .properties()
+                    .expression(mappingCtrExprNode);
+        } else if (kind == SyntaxKind.ASSIGNMENT_STATEMENT) {
+            startNode(FlowNode.Kind.UPDATE_DATA)
+                    .properties()
+                    .expression(mappingCtrExprNode)
+                    .variable(((AssignmentStatementNode) mappingCtrExprNode.parent()).varRef());
+        }
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java:505]: Shall we extract this to a method and use it for both the methods?

Comment (nipunayf) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java:505]: Will address the following in another PR.

**Pull request merged**